test/exceptions/src/assert_fail.d

```diff
@@ -444,6 +444,52 @@ void testException()
     test(MayThrow(0), MayThrow(1), `Some message != <toString() failed: "Error", called on MayThrow(1)>`);
 }
 
+void testOverlappingFields()
+{
+    static struct S
+    {
+        union
+        {
+            double num;
+            immutable(char)[] name;
+        }
+    }
+
+    test(S(1.0), S(2.0), "S(<overlapped field>, <overlapped field>) != S(<overlapped field>, <overlapped field>)");
```

Comment:
Shouldn't this be `<overlapped field>` ? As in, print only one entry per union ?